TypeScript compiler 4.5 reports type definition not found

I've created a minimal reproducible example for this issue, see [equt-playground/typescript-4.5-vite-plugin-pages](https://github.com/equt-playground/typescript-4.5-vite-plugin-pages).

The branch `4.4` using `typescript@4.4` reports no error on a `tsc` check, while after switching to the latest `typescript@4.5` (branch `4.5`), it will complain that `vite-plugin-pages/client` not found.

```
> tsc

error TS2688: Cannot find type definition file for 'vite-plugin-pages/client'.
  The file is in the program because:
    Entry point of type library 'vite-plugin-pages/client' specified in compilerOptions

  tsconfig.json:3:37
    3     "types": ["vue", "vite/client", "vite-plugin-pages/client", "@types/node"],
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~
    File is entry point of type library specified here.


Found 1 error.
```

It's kinda weird, since `vite` seems to use the same approach but results in no error. So I belive that this should be somehow caused by `vite-plugin-pages`.

BTW, remember to do an installation after switching the branch.

--- a/package.json
+++ b/package.json
@@ -18,6 +18,12 @@
       "require": "./dist/index.js",
       "import": "./dist/index.mjs",
       "types": "./dist/index.d.ts"
+    },
+    "./client": {
+      "types": "./client.d.ts"
+    },
+    "./client-react": {
+      "types": "./client-react.d.ts"
     }
   },
   "main": "dist/index.js",

